api4/team_test.go

```diff
@@ -459,6 +459,68 @@ func TestPatchTeamSanitization(t *testing.T) {
 	})
 }
 
+func TestUpdateTeamPrivacy(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+	Client := th.Client
+
+	createTeam := func(teamType string, allowOpenInvite bool) *model.Team {
+		team := &model.Team{
+			DisplayName:     teamType + " Team",
+			Description:     "Some description",
+			CompanyName:     "Some company name",
+			AllowOpenInvite: allowOpenInvite,
+			InviteId:        model.NewId(),
+			Name:            "aa-" + model.NewRandomTeamName() + "zz",
+			Email:           "success+" + model.NewId() + "@simulator.amazonses.com",
+			Type:            teamType,
+		}
+		team, _ = Client.CreateTeam(team)
+		return team
+	}
+
+	teamPublic := createTeam(model.TEAM_OPEN, true)
+	teamPrivate := createTeam(model.TEAM_INVITE, false)
+
+	teamPublic2 := createTeam(model.TEAM_OPEN, true)
+	teamPrivate2 := createTeam(model.TEAM_INVITE, false)
+
+	tests := []struct {
+		name           string
+		team           *model.Team
+		privacy        string
+		wantErr        bool
+		wantType       string
+		wantOpenInvite bool
+	}{
+		{name: "bad privacy", team: teamPublic, privacy: "blap", wantErr: true, wantType: model.TEAM_OPEN, wantOpenInvite: true},
+		{name: "bad team", team: &model.Team{Id: model.NewId() + "nope"}, privacy: "public", wantErr: true, wantType: model.TEAM_OPEN, wantOpenInvite: true},
+		{name: "public to private", team: teamPublic, privacy: "private", wantErr: false, wantType: model.TEAM_INVITE, wantOpenInvite: false},
+		{name: "private to public", team: teamPrivate, privacy: "public", wantErr: false, wantType: model.TEAM_OPEN, wantOpenInvite: true},
+		{name: "public to public", team: teamPublic2, privacy: "public", wantErr: false, wantType: model.TEAM_OPEN, wantOpenInvite: true},
+		{name: "private to private", team: teamPrivate2, privacy: "private", wantErr: false, wantType: model.TEAM_INVITE, wantOpenInvite: false},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			resp := Client.UpdateTeamPrivacy(test.team.Id, test.privacy)
+			if test.wantErr {
+				CheckBadRequestStatus(t, resp)
+				return
+			} else {
+				CheckNoError(t, resp)
+				CheckOKStatus(t, resp)
+			}
+
+			team, err := th.App.GetTeam(test.team.Id)
+			require.Nil(t, err)
+
+			require.Equal(t, test.wantType, team.Type)
+			require.Equal(t, test.wantOpenInvite, team.AllowOpenInvite)
+		})
+	}
+}
```

Comment:
Can we add tests for the following?
- related to permission (with / without permission to manage a team - forbidden), and
- logged in or not logged in user - unauthorized

Reply:
Unit tests added here https://github.com/mattermost/mattermost-server/pull/14287/commits/045457a23d4112cf390b6ae5f244a0ae194f1ec8   Thanks!